### PR TITLE
#208: Post-processing not triggered when resolving proxies without type inference

### LINKED ISSUE #208: Post-processing not triggered when resolving proxies without type inference
Post-processing of an N4JSResource (aka AST traversal) should be invoked when the first proxy is resolved, as stated in the design document Section 7.5.2.). However, this is not working in all cases.

To reproduce:
```
// Xtend code

class TempTest extends AbstractN4JSTest {

	@Test
	def void test() throws Exception {
		val script = '''
			let a = 'hello';
			console.log(a);
		'''.parseN4js;

		val res = script.eResource as N4JSResource;
		val identRef = script.eAllContents.filter(IdentifierRef).head;

		identRef.id; // should trigger post-processing, but doesn't
		assertTrue(res.isFullyProcessed);
	}
}
```

AC:

1. investigate.
2. fix.
3. add tests for all expected ways of triggering post-processing!

### PR DESCRIPTION
**Pull request by mor-n4**
fixes #208.

### CODE REVIEW

**Comment by mor-n4** [comment]
Hi @qtran-n4,
builds are green. Please review!

File: tests/org.eclipse.n4js.lang.tests/src/org/eclipse/n4js/tests/scoping/ComposedMemberCachingTest.xtend [hunk 1]
```diff
@@ -150,19 +152,39 @@ class ComposedMemberCachingTest extends AbstractN4JSTest {
...
 		val scope2 = memberScopingHelper.createMemberScope(receiverTypeRef, propAccess, false, true);
 		scope2.getAllElements();
 		assertTrue("scoping should have added several members to composed member cache",
 			script.module.composedMemberCaches.head.cachedComposedMembers.size>1); // at time of writing, 16 members were added (but exact count might change over time)
 	}
 
+	def private Script parseAndPostProcessAndDeleteCachedComposedMembers(CharSequence code) {
+		val script = code.parseAndValidateSuccessfully;
+		assertTrue((script.eResource as N4JSResource).isFullyProcessed); // ensure validation did trigger post-processing
+		// While resolving the proxies during validation, composed members will have been added to the composed member
```
Comment (qtran-n4) [inline tests/org.eclipse.n4js.lang.tests/src/org/eclipse/n4js/tests/scoping/ComposedMemberCachingTest.xtend]: It took me a while to understand this comment. Probably it is better to say something like "Here, we are are going to call scoping manually and test that composed member cache is created correctly"

File: tests/org.eclipse.n4js.lang.tests/src/org/eclipse/n4js/tests/postprocessing/PostProcessingTriggerTest.xtend [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import org.eclipse.n4js.typesystem.N4JSTypeSystem
+import org.eclipse.n4js.utils.emf.ProxyResolvingResource
+import org.junit.Test
+
+import static extension org.eclipse.n4js.typesystem.RuleEnvironmentExtensions.*
+
+/**
+ * Tests the various ways of triggering post-processing.
+ */
+class PostProcessingTriggerTest extends AbstractN4JSTest {
```
Comment (qtran-n4) [inline tests/org.eclipse.n4js.lang.tests/src/org/eclipse/n4js/tests/postprocessing/PostProcessingTriggerTest.xtend]: Nice test cases!

**Comment by mor-n4** [comment]
thanks for the review!
builds green, merging now...

**Pull request merged**